tajo-core/src/main/java/org/apache/tajo/master/TajoMasterClientService.java

```diff
@@ -428,9 +426,7 @@ public GetQueryListResponse getFinishedQueryList(RpcController controller, GetQu
           infoBuilder.setState(queryInfo.getQueryState());
           infoBuilder.setQuery(queryInfo.getSql());
           infoBuilder.setStartTime(queryInfo.getStartTime());
-          long endTime = (queryInfo.getFinishTime() == 0) ?
-              System.currentTimeMillis() : queryInfo.getFinishTime();
-          infoBuilder.setFinishTime(endTime);
+          infoBuilder.setFinishTime(System.currentTimeMillis());
```

Comment:
It seems like last request time..
I suggest that the {{isFinishState}} branch in {{QueryInProgress.hearbeart}} sets the finish time